Can not synthesize CloudFormation template for fargate-service-with-auto-scaling

## :bug: Bug Report

### What is the problem?

Can not synthesize CloudFormation template for fargate-service-with-auto-scaling

### Reproduction Steps
```bash

git clone https://github.com/aws-samples/aws-cdk-examples.git
cd aws-cdk-examples/typescript/ecs/fargate-service-with-auto-scaling
npm install
npm run build
```

### Verbose Log

```
> fargate-service-with-auto-scaling@1.0.0 build /home/ec2-user/environment/aws-cdk-examples/typescript/ecs/fargate-service-with-auto-scaling
> tsc

index.ts:17:7 - error TS2345: Argument of type '{ cluster: Cluster; image: RepositoryImage; }' is not assignable to parameter of type 'NetworkLoadBalancedFargateServiceProps'.
  Object literal may only specify known properties, and 'image' does not exist in type 'NetworkLoadBalancedFargateServiceProps'.

17       image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample")
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! fargate-service-with-auto-scaling@1.0.0 build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the fargate-service-with-auto-scaling@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ec2-user/.npm/_logs/2019-10-07T18_43_55_681Z-debug.log
```

### Environment

  - **CDK CLI Version:** 1.12.0 (build 923055e)
  - **OS:** Linux ip-172-31-26-69 4.14.143-91.122.amzn1.x86_64 #1 SMP Wed Sep 11 00:43:34 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
  - **Language:**  TypeScript

--- a/python/ecs/ecs-load-balanced-service/app.py
+++ b/python/ecs/ecs-load-balanced-service/app.py
@@ -28,7 +28,9 @@ class BonjourECS(core.Stack):
             self, "Ec2Service",
             cluster=cluster,
             memory_limit_mib=512,
-            image=ecs.ContainerImage.from_registry("amazon/amazon-ecs-sample")
+            task_image_options={
+                'image': ecs.ContainerImage.from_registry("amazon/amazon-ecs-sample")
+            }
         )
 
         core.CfnOutput(

--- a/python/ecs/fargate-load-balanced-service/app.py
+++ b/python/ecs/fargate-load-balanced-service/app.py
@@ -26,7 +26,9 @@ class BonjourFargate(core.Stack):
         fargate_service = ecs_patterns.NetworkLoadBalancedFargateService(
             self, "FargateService",
             cluster=cluster,
-            image=ecs.ContainerImage.from_registry("amazon/amazon-ecs-sample")
+            task_image_options={
+                'image': ecs.ContainerImage.from_registry("amazon/amazon-ecs-sample")
+            }
         )
 
         core.CfnOutput(

--- a/python/ecs/fargate-service-with-autoscaling/app.py
+++ b/python/ecs/fargate-service-with-autoscaling/app.py
@@ -26,7 +26,9 @@ class AutoScalingFargateService(core.Stack):
         fargate_service = ecs_patterns.NetworkLoadBalancedFargateService(
             self, "sample-app",
             cluster=cluster,
-            image=ecs.ContainerImage.from_registry("amazon/amazon-ecs-sample")
+            task_image_options={
+                'image': ecs.ContainerImage.from_registry("amazon/amazon-ecs-sample")
+            }
         )
 
         # Setup AutoScaling policy

--- a/typescript/ecs/ecs-load-balanced-service/index.ts
+++ b/typescript/ecs/ecs-load-balanced-service/index.ts
@@ -21,7 +21,9 @@ class BonjourECS extends cdk.Stack {
     const ecsService = new ecs_patterns.NetworkLoadBalancedEc2Service(this, "Ec2Service", {
       cluster,
       memoryLimitMiB: 512,
-      image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
+      taskImageOptions: {
+        image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
+      }
     });
 
     // Output the DNS where you can access your service

--- a/typescript/ecs/fargate-load-balanced-service/index.ts
+++ b/typescript/ecs/fargate-load-balanced-service/index.ts
@@ -15,7 +15,9 @@ class BonjourFargate extends cdk.Stack {
     // Instantiate Fargate Service with just cluster and image
     const fargateService = new ecs_patterns.NetworkLoadBalancedFargateService(this, "FargateService", {
       cluster,
-      image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
+      taskImageOptions: {
+        image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
+      },
     });
 
     // Output the DNS where you can access your service

--- a/typescript/ecs/fargate-service-with-auto-scaling/index.ts
+++ b/typescript/ecs/fargate-service-with-auto-scaling/index.ts
@@ -14,7 +14,9 @@ class AutoScalingFargateService extends cdk.Stack {
     // Create Fargate Service
     const fargateService = new ecs_patterns.NetworkLoadBalancedFargateService(this, 'sample-app', {
       cluster,
-      image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample")
+      taskImageOptions: {
+        image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample")
+      },
     });
 
     // Setup AutoScaling policy

--- a/typescript/ecs/fargate-service-with-local-image/index.ts
+++ b/typescript/ecs/fargate-service-with-local-image/index.ts
@@ -12,13 +12,15 @@ const stack = new cdk.Stack(app, 'FargateServiceWithLocalImage');
 const vpc = new ec2.Vpc(stack, 'MyVpc', { maxAzs: 2 });
 const cluster = new ecs.Cluster(stack, 'Cluster', { vpc });
 
-// Instantiate Fargate Service with a cluster and a local image that gets 
+// Instantiate Fargate Service with a cluster and a local image that gets
 // uploaded to an S3 staging bucket prior to being uploaded to ECR.
 // A new repository is created in ECR and the Fargate service is created
 // with the image from ECR.
 new ecs_patterns.NetworkLoadBalancedFargateService(stack, "FargateService", {
   cluster,
-  image: ecs.ContainerImage.fromAsset(path.resolve(__dirname, 'local-image'))
+  taskImageOptions: {
+    image: ecs.ContainerImage.fromAsset(path.resolve(__dirname, 'local-image'))
+  }
 });
 
 app.synth();

